### PR TITLE
Fix/#231 normalize figure alt text

### LINKED ISSUE #231: Article editor, formatted text causes caption fragmentation after markdown export when used in figcaption
* PR target branch: `feature/articles`

# Steps to reproduce

* drop an image in the editor
* click the image, enter a caption at the bottom
* select a section of the caption
* press `CTRL` + `i`
* click on `preview & mint` at the bottom, which logs the markdown to the console

# Demo

Image

```markdown
![Some text, ](blob:http://localhost:3000/aea137a2-6cfc-4c21-8bdb-e09990cf05f2)
```

The italic node cuts the text in the caption

# Probable location

https://github.com/fxhash/fxhash-website/blob/f2b4294f7ea85d9a6f74ae8fd6f7651e5d132bc3/src/components/NFTArticle/processor.ts#L219

# Solution

- [x] remove text styling on the figcaption (no menu, no shortcut)
- [x] enforce figcaptions to only be a simple text node through normalization (https://docs.slatejs.org/concepts/11-normalizing)

### PR DESCRIPTION
**Pull request by maerzhase**
__This PR requires #250__

@ciphrd I introduced a new boolean `hideFloatingInlineMenu` to the `BlockDefinitions`. This way we can control which nodes will actually show the inline style menu. Further more i've also added the normalisation of the figcaption into the `ImagePlugin`. 

- fix #231

### CODE REVIEW

**Review by ciphrd — CHANGES_REQUESTED** [review]
Great job!

One remark: it's still possible to paste an image inside the figcaption, the normalization should also prevent that (and not only that but also any non-text leafs)

image

File: src/components/NFTArticle/SlateEditor/Plugins/SlateImagePlugin.ts [hunk 1]
```diff
@@ -88,7 +103,13 @@ export const withImages = (
...
             })
           }
-        }
+	} else {
+	  // any child that is not the first image in the figure moved after the figure
+	  if (!(child.type === 'image' && childPath[childPath.length - 1] === 0)) {
+	    Transforms.liftNodes(editor, {at: childPath})
+	    return;
+	  }
+	}
```
Comment (maerzhase) [inline src/components/NFTArticle/SlateEditor/Plugins/SlateImagePlugin.ts:112]: @ciphrd I added this line which moves any child beside the first image after the figure. We could also change the `liftNodes` to `removeNodes` but imho the UX is better when e.g. the images are actually pasted into the editor. What do you say?

Comment (ciphrd) [reply src/components/NFTArticle/SlateEditor/Plugins/SlateImagePlugin.ts]: Yes I believe this is a great UX, perfect :)

**Review by ciphrd — APPROVED** [review]
LGTM!

Merging

**Pull request merged**